### PR TITLE
fix: group messages date | design changes (WPB-1733)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1733" title="WPB-1733" target="_blank">StoryWPB-1733</a>  Timestamps – grouping functionality for dates
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Edge case when message is the first and only message
- Edge case when its the last message
- Design on divider margins

### Solutions

- Some design changes (margin on divider)
- Addition of edge case when its only 1 message or the last message
- Addition of 45min gap handling
- Remove date from the right side of Message Item (instead of [date|time] -> [time])

### Dependencies (Optional)

Needs releases with:

- [X] #2878 
- [X] #2964 

Screenshot 2024-05-06 at 12 00 33
Screenshot 2024-05-06 at 12 00 08

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt [hunk 1]
```diff
@@ -1104,6 +1103,16 @@ fun MessageList(
                         isSelectedMessage = (message.header.messageId == selectedMessageId),
                         isInteractionAvailable = interactionAvailability == InteractionAvailability.ENABLED
                     )
+
+                    if ((index == 0 && lazyPagingMessages.itemCount == 1) || (index + 1) == lazyPagingMessages.itemCount) {
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt]: Just to make it a bit more readable:

```suggestion
                    val isTheOnlyItem = index == 0 && lazyPagingMessages.itemCount == 1
                    val isTheLastItem = (index + 1) == lazyPagingMessages.itemCount
                    if (isTheOnlyItem || isTheLastItem) {
```

File: app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt [hunk 2]
```diff
@@ -181,10 +177,7 @@ sealed interface MessageDateTimeGroup {
 
 fun String.uiMessageDateTime(): String? = this
     .serverDate()?.let { serverDate ->
-        when (DateUtils.isToday(serverDate.time)) {
-            true -> messageTimeFormatter.format(serverDate)
-            false -> messageDateTimeFormatter.format(serverDate)
-        }
+        messageTimeFormatter.format(serverDate)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt:183]: What is the goal of this change? Not sure if this is related to the current PR description.

**Comment by AndroidBob** [comment]
Build [4634](https://10.10.124.134/job/AR-build-pipeline/4634/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9604328)(9604328)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9604328)(9604328)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt (hunk 2 above)
Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/util/DateTimeUtil.kt:183]: Forgot to mention on the description, sorry :(

This change is for removing the date and only showing the time on the right side of the message item, as we are adding dividers with dates, there is no reason to have it on the message item as well.

based on the designs and followed up with our team designer

**Review by vitorhugods — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4655](https://10.10.124.134/job/AR-build-pipeline/4655/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9612212)(9612212)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9612212)(9612212)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4663](https://10.10.124.134/job/AR-build-pipeline/4663/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9613080)(9613080)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9613080)(9613080)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]

**Pull request merged**